qubesmanager/settings.py

```diff
@@ -1270,32 +1270,32 @@ def __apply_devices_tab__(self):
             for dev in new_devs:
                 if dev not in old_devs:
                     options = {}
-                    if dev.ident in self.new_strict_reset_list:
+                    if dev.port_id in self.new_strict_reset_list:
                         options['no-strict-reset'] = True
-                    ass = devices.DeviceAssignment(
-                        self.vm.app.domains['dom0'],
-                        dev.ident, devclass='pci',
-                        attach_automatically=True, required=True,
+                    ass = device_protocol.DeviceAssignment(
+                        device_protocol.Port(
+                            self.vm.app.domains['dom0'], dev.port_id, 'pci'),
+                        mode='required',
```

Comment:
This does not match new DeviceAssignment constructor after https://github.com/QubesOS/qubes-core-admin-client/pull/304/commits/8772108b263a3379a3c53cf8b6cb14576889bb7d#diff-3c9f34f047553a6c2bc231786134cfece1c2d94ba3931a216f6a91840e130e95L860-R1015

Maybe you want to use the DeviceAssignment.new helper here?